### ISSUE TITLE
Bitstream CICP data loss in decode_image_planar

### ISSUE THREAD

**Issue by 0xC0000054** [strukturag/libheif#346]
When loading an image that stores the CICP values in the bitstream, those values will be lost after the `convert_image` call.
This causes corrupted colors when using the Identity matrix coefficient with an ICC color profile.

I think the fix would be to add an `else if` statement that uses the bitstream NCLX box if the HEIF metadata does not contain one.
Something like:

```C++
if (colorProfileInMetadata) {
      img->set_color_profile_nclx(colorProfileInMetadata);
    }
else if (nclx && !img->get_color_profile_nclx()) {
     img->set_color_profile_nclx(nclx);
    }
```

**Comment by farindk** [comment]
Do you have a test file at hand with CICP only in the bitstream?

**Comment by 0xC0000054** [comment]
>  This causes corrupted colors when using the Identity matrix coefficient with an ICC color profile.

After further investigation it looks like the issue is due to libde265 reading a matrix coefficient value of `2` from the VUI header.
The x265 encoder is setting a matrix coefficient value of `0` in its encoder parameters, so I am not sure what is causing the problem.

The attached file is a HEIC image that has a embedded ICC profile, the VUI header should have the following CICP values:

Color primaries: 1
Transfer characteristics: 13
Matrix coefficient: 0
Full range: true

[Embedded ICC Profile.zip](https://github.com/strukturag/libheif/files/5299019/Embedded.ICC.Profile.zip)

**Comment by 0xC0000054** [comment]
Fixed with strukturag/libde265@60d682d
